object-manipulation-3/main.js

```diff
@@ -0,0 +1,117 @@
+// console.log('Lodash is loaded:', typeof _ !== 'undefined');
+
+// declare object variable players that hold 4 objects with each holding a name property as string...
+// and hand property as empty array literal
+var players = {
+  player1: { name: 'Neil', hand: [] },
+  player2: { name: 'Breen', hand: [] },
+  player3: { name: 'Tommy', hand: [] },
+  player4: { name: 'Wiseau', hand: [] }
+};
+
+// declare variable array literals holding ranks and suits list in order to create a new deck
+var arrayRank = ['ace', 2, 3, 4, 5, 6, 7, 8, 9, 10, 'jack', 'queen', 'king'];
+var arraySuit = ['clubs', 'diamonds', 'hearts', 'spades'];
+
+// create a new deck
+function createDeckArray(rank, suit) {
+  var deck = [];
+  for (var i = 0; i < suit.length; i++) {
+    for (var n = 0; n < rank.length; n++) {
+      // tempObj holds a temporary object that holds 2 properties and will be...
+      // ... appended to deck
+      var tempObj = {};
+      tempObj[suit[i]] = rank[n];
```

Comment:
This isn't an object that holds the two properties `rank` and `suit`, this is an object that hold _one_ property, where the name of the property is the suit:

```js
/* Example card where the suit is 'hearts' and the rank is 6 */
// Expected tempObj:
{ rank: 6, suit: 'hearts' }
// The actual tempObj this code is creating:
{ hearts: 6 }
```

---

![image](https://user-images.githubusercontent.com/16143771/153051163-1964697d-c5d1-4ab6-927d-5c3dee189284.png)